File: backend/app/schemas/card.py
from pydantic import BaseModel


class CardSummary(BaseModel):
    id: int
    name_ko: str
    name_en: str
    arcana: str
    suit: str | None = None
    image_url: str | None = None

    model_config = {"from_attributes": True}


class CardDetail(CardSummary):
    number: int | None = None
    description: str | None = None

File: backend/src/main/java/com/arcana/backend/entity/Card.java
package com.arcana.backend.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "cards")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Card {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "name_ko", nullable = false, length = 100)
    private String nameKo;

    @Column(name = "name_en", nullable = false, length = 100)
    private String nameEn;

    @Column(nullable = false, length = 10)
    private String arcana;

    @Column(length = 50)
    private String suit;

    @Column
    private Integer number;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "image_url", length = 500)
    private String imageUrl;
}
